basics/section_1-2.rb

```diff
@@ -0,0 +1,16 @@
+# Section 1-2: Lets code the quiz!
+
+address = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+# address variable
+
+p address
```

Comment:
Why use `p`? What are the other ways of displaying to console?
Why are the pros & cons of the others?

Reply:
`p` was used here for testing purposes. Other ways of display to the console can be done with `puts` or `print`.
- `print` prints inline without placing the next statement on the next line, can be useful for scenarios when we want to display a divider with multiple `-` onto the console.
- `puts` is similar to print but prints on a new line, it also prints each element of the array instead of printing the array itself. Can be useful for displaying content for the user without showing `""` for strings and `[]` for arrays.
-   `p` prints & returns whatever argument you give it. Useful for debugging when we want to see exactly what is being fed in.

Reply:
I am asking, why use `p` instead of the others?

Reply:
With `p`, we have a better indicator of the class of the object being printed than the others.

Reply:
Why do you say so? Can you give an example?

Reply:
`p` "Hello world" prints `"Hello world"` whereas `puts` "Hello world" prints `Hello world`.